### ISSUE TITLE
1 km square allocation map

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/PoMS#83]
https://test-ceh-poms.pantheonsite.io/square-volunteer-pantrap-statuses

**Map symbol colours:**
At the moment unallocated squares are semi-transparent red, changing to solid red when allocated. Can this be changed so that unallocated squares are semi-transparent red and allocated squares are semi-transparent dark blue.

I think you were also intending to vary the colour depending on whether the squares have been surveyed, but I don't think that is necessary - the key message is whether or not they are allocated and that's all that we need to distinguish on the map.

**Grid columns:**
We don't need to include the Square ID column.

I assume the "Count" column is a count of the number of samples allocated to that square? We'll need to filter this report to show data for the current year only.

The grid contains 78 squares, but only 75 of those are in use (the other three were used in previous years, and may have data associated with them, but have been dropped and should not be available for future data entry). What's the best way to exclude the three that we don't want - can I mark them in the warehouse as not available to the PoMS website? Or do we need an additional custom location attribute for "square no longer active"?

**Map clicks:**
As with the other maps, it would be good if we can enable clicking on the map dots to highlight the relevant row in the grid and vice-versa.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Just working on report for these changes.
I can see that I think the grid clicking is being broken by the same problem as the data entry forms have, the javascript is not loading properly.
In terms of the exclusions.
I think this may need extra budget, as I will need to change the whole system because the squares need excluding from the data entry forms also, affects other reports also. Is this urgent? we could leave to end and see if budget is left.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Hi Martin,
I need to check what is meant by "current year".
Do you mean calendar year, or do you mean the past year from the date of viewing?
So the former of these will show a count of 0 as soon as you reach 1st Jan, but the second will have a count that changes day to day.

**Comment by kitenetter** [comment]
@andrewvanbreda 
I mean the calendar year, so that if you access the page on any date in 2021 it shows only the surveys from 2021, but as soon as we hit 1 Jan 2022 it 'resets' and only shows data for 2022.

**Comment by kitenetter** [comment]
@andrewvanbreda 
Regarding the square exclusions, am I correct in thinking that the list of squares that you are pulling in to the report is defined by the location type, as discussed [here](https://github.com/BiologicalRecordsCentre/PoMS/issues/14#issuecomment-787892797)?

If so, will it work if I edit the three squares that we don't want, and give them a different location type?

**Comment by andrewvanbreda** [comment]
@kitenetter 
Yes, actually yes this latter idea is good. I like that idea.
Seems so obvious now, kicking myself didn't think of that.
Let me know which ones and I will switch them

**Comment by kitenetter** [comment]
@andrewvanbreda I've edited the location type for the three squares that are no longer needed:

![image](https://user-images.githubusercontent.com/16039111/116557394-4e0f9f80-a8f6-11eb-8b0d-4e9ecd198707.png)

Hope that's the right thing to do, let me know if anything needs to change.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Yes this is great idea, and looks like you have also done right thing there also.

**Comment by kitenetter** [comment]
Additional request:
Add "Location code" from the locations table into the grid (to replace "Square ID").

**Comment by andrewvanbreda** [comment]
Report not deployed yet, but have so far committed following fixes

1. Allocated semi-transparent dark blue, unallocated semi-transparent red.
2. Removes colour variation depending on count
3. Square ID removed
4. Code added instead
5. Count is now for calendar year only with automatic switch to next year
6. (Martin has already fixed the 3 unused squares)

Just need to fix the map click, should be straightforward enough.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Hi Biren,
Are you able to pull the squares_summary_mappable report in indicia-reports PoMS.
It is a live report, but I don't think there will be an issue as tested on my machine and none of the parameters have changed.

**Comment by BirenRathod** [comment]
@andrewvanbreda I have pulled that report now.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Thanks Biren. That worked fine, although am noticing a problem with the count field that I am not seeing on my own machine (this would not be a new problem).
Could you pull it again please as have tried to add a distinct to the field.
Cheers.

**Comment by BirenRathod** [comment]
@andrewvanbreda I pulled it again.

**Comment by andrewvanbreda** [comment]
@BirenRathod Thanks Biren,
Yes that looks fixed now :)


AVB: Note to myself, just the grid clicking to fix now

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Could you pull squares_summary_mappable PoMS report on live whenever you get a chance, cheers. Should be ok as tested on my machine and have preconfigured the form to take into account the change.
Cheers

**Comment by BirenRathod** [comment]
@andrewvanbreda I just pulled the repository but that report didn't update. Any idea?

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Hi Biren, 
That was an odd one. I checked Github and you are right, my commit wasn't there, even though the repository on my local machine was acting like it had been pushed successfully (git diff, git status showed nothing, even though my change was present in the file).
I did a hard reset and I could see my change again, so have recommitted and pushed.
I can now see it in Github
